common/models/transducer/topology.py

```diff
@@ -0,0 +1,43 @@
+from typing import Callable, List
+from returnn.tf.util.data import Data
+from tensorflow.python.framework.ops import Tensor
+
+from .loss import rnnt_loss, rnnt_loss_out_type, rnnt_tf_loss, rna_tf_loss, rna_loss_out_type
+from .alignment import rnnt_alignment, rnnt_alignment_out_type, rna_alignment, rna_alignment_out_type
+from tensorflow import Tensor
+
+
+class Topology:
+  def __init__(self, name: str,
+               loss: Callable[[List[Data], Tensor]],
```

Comment:
This looks wrong. Shouldn't it be 
```suggestion
               loss: Callable[[List[Data]], Tensor],
```
Did you actually verify this? E.g. in PyCharm, or MyPy or whatever. PyCharm should give you warnings when you specify it incorrectly, or when you pass the wrong type. And you can also inspect whether it actually has set the correct type.

Reply:
Also, despite these basic wrong usage, I think it is further incorrect. It doesn't get a `List[Data]` as argument. The argument is the named arg `source`, which is not a list but another function i -> Data.